### PR TITLE
Merge different languages for language stats

### LINKED ISSUE #24896: Unable to update language stats for ID and stats queue indexer.Index(1424) failed
### Description

I mirrored [the repository](https://github.com/itwanger/paicoding),Every time I restart gitea will report some error.

1. Unable to update language stats for ID
2. stats queue indexer.Index(1424) failed

image

### Gitea Version

v1.20+

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

Ubuntu

### How are you running Gitea?

./gitea web

### Database

MySQL

### PR DESCRIPTION
**Pull request by wxiaoguang**
Fix #24896

If users set different languages by `linguist-language`, the `stats` map could be: `java: 100, Java: 200`.

Language stats are stored as case-insensitive in database and there is a unique key.

So, the different language names should be merged to one unique name: `Java: 300`

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: modules/git/repo_language_stats_test.go [hunk 1]
```diff
@@ -30,3 +30,17 @@ func TestRepository_GetLanguageStats(t *testing.T) {
...
+		"PHP":    1,
+		"python": 10,
+		"JAVA":   700,
+	}, mergeLanguageStats(map[string]int64{
+		"PHP":    1,
+		"python": 10,
+		"Java":   100,
+		"java":   200,
+		"JAVA":   400,
+	}))
```
Comment (silverwind) [inline modules/git/repo_language_stats_test.go:45]: Is is normal in our go tests to write the expected value first? Being used to jest/vitest `expect(value).toEqual(expected)` order, this reverse order looks odd to me.

**Review by silverwind — APPROVED** [review]

File: modules/git/repo_language_stats_test.go (hunk 1 above)
Comment (wxiaoguang) [reply modules/git/repo_language_stats_test.go:45]: In most cases, the expected value always comes first.

image

Comment (silverwind) [reply modules/git/repo_language_stats_test.go:45]: I see.

**Comment by wxiaoguang** [comment]
backport ->  Merge different languages for language stats #24921

**Pull request merged**